pkg/cache/clusterqueue_test.go

```diff
@@ -98,206 +98,281 @@ func TestFitInCohort(t *testing.T) {
 	cases := map[string]struct {
 		request            resources.FlavorResourceQuantitiesFlat
 		wantFit            bool
-		cq                 *ClusterQueueSnapshot
+		flavorNames        []string
+		usage              resources.FlavorResourceQuantitiesFlat
+		clusterQueue       []*kueue.ClusterQueue
 		enableLendingLimit bool
 	}{
 		"full cohort, empty request": {
-			request: resources.FlavorResourceQuantitiesFlat{},
-			wantFit: true,
-			cq: &ClusterQueueSnapshot{
-				Name: "CQ",
-				Cohort: &CohortSnapshot{
-					Name: "C",
-					RequestableResources: resources.FlavorResourceQuantitiesFlat{
-						{Flavor: "f1", Resource: corev1.ResourceCPU}:    5,
-						{Flavor: "f1", Resource: corev1.ResourceMemory}: 5,
-						{Flavor: "f2", Resource: corev1.ResourceCPU}:    5,
-						{Flavor: "f2", Resource: corev1.ResourceMemory}: 5,
-					}.Unflatten(),
-					Usage: resources.FlavorResourceQuantitiesFlat{
-						{Flavor: "f1", Resource: corev1.ResourceCPU}:    5,
-						{Flavor: "f1", Resource: corev1.ResourceMemory}: 5,
-						{Flavor: "f2", Resource: corev1.ResourceCPU}:    5,
-						{Flavor: "f2", Resource: corev1.ResourceMemory}: 5,
-					}.Unflatten(),
-				},
-				ResourceGroups: nil,
+			request:     resources.FlavorResourceQuantitiesFlat{},
+			wantFit:     true,
+			flavorNames: []string{"f1", "f2"},
+			usage: resources.FlavorResourceQuantitiesFlat{
+				{Flavor: "f1", Resource: corev1.ResourceCPU}:    5_000,
+				{Flavor: "f1", Resource: corev1.ResourceMemory}: 5,
+				{Flavor: "f2", Resource: corev1.ResourceCPU}:    5_000,
+				{Flavor: "f2", Resource: corev1.ResourceMemory}: 5,
+			},
+			clusterQueue: []*kueue.ClusterQueue{
+				utiltesting.
+					MakeClusterQueue("CQ").
+					ResourceGroup(
+						utiltesting.MakeFlavorQuotas("f1").
+							ResourceQuotaWrapper(corev1.ResourceCPU).NominalQuota("5").Append().
+							FlavorQuotas,
+					).
+					ResourceGroup(
+						utiltesting.MakeFlavorQuotas("f1").
+							ResourceQuotaWrapper(corev1.ResourceMemory).NominalQuota("5").Append().
+							FlavorQuotas,
+					).
+					ResourceGroup(
+						utiltesting.MakeFlavorQuotas("f2").
+							ResourceQuotaWrapper(corev1.ResourceCPU).NominalQuota("5").Append().
+							FlavorQuotas,
+					).
+					ResourceGroup(
+						utiltesting.MakeFlavorQuotas("f2").
+							ResourceQuotaWrapper(corev1.ResourceMemory).NominalQuota("5").Append().
+							FlavorQuotas,
+					).
```

Comment:
Are the tests passing? Kueue constrains each flavor and each resource to only belong to a single resource group (see https://github.com/kubernetes-sigs/kueue/pull/2592#discussion_r1675774465). Perhaps it is only enforced by the webhook, however

Please update tests to meet this constraint

Reply:
fix: https://github.com/kubernetes-sigs/kueue/pull/2655/commits/cd25e816307aea3916f6050214d3bd880dc164e1

Reply:
This is _nearly_ right. However, both flavors should be in the same `ResourceGroup`, as they cover the same resources (cpu/mem).

For this entire test suite, there should just be 1 ResourceGroup, with 1-2 flavors each.

Reply:
fix: https://github.com/kubernetes-sigs/kueue/pull/2655/commits/e0b4e745f7f498b0804b80fc253517e9c74026d6